torch/_dynamo/convert_frame.py

```diff
@@ -1630,6 +1631,21 @@ def __call__(
     ) -> ConvertFrameReturn: ...
 
 
+def should_skip_due_to_torch_dispatch_mode() -> bool:
+    # fast check
+    if not is_in_torch_dispatch_mode(include_infra_modes=False):
+        return False
+
+    mode_stack = _get_current_dispatch_mode_stack()
+    for mode in mode_stack:
+        if mode.ignore_compile_internals():
+            continue
+        if mode.is_infra_mode():
+            continue
+        return True
```

Comment:
hmm do you mind explaining what the behavior is when you have multiple modes active, and one has `ignore_compile_internals` set but the other does not?